spec/forms/state_file/md_had_health_insurance_form_spec.rb

```diff
@@ -0,0 +1,97 @@
+require "rails_helper"
+
+RSpec.describe StateFile::MdHadHealthInsuranceForm do
+  let(:intake) { create :state_file_md_intake }
+
+  describe "validations" do
+    context "had_hh_member_without_health_insurance is nil" do
+      it "is invalid" do
+        form = described_class.new(intake, {
+          had_hh_member_without_health_insurance: nil
+        })
+        expect(form.valid?).to eq false
+        expect(form.errors[:had_hh_member_without_health_insurance]).to include "Can't be blank."
+      end
+    end
+
+    context "had_hh_member_without_health_insurance is no" do
+      it "is valid" do
+        form = described_class.new(intake, {
+          had_hh_member_without_health_insurance: "no"
+        })
+        expect(form.valid?).to eq true
+      end
+    end
+
+    context "had_hh_member_without_health_insurance" do
+      it "is invalid when authorize_sharing_of_health_insurance_info is nil" do
+        form = described_class.new(intake, {
+          had_hh_member_without_health_insurance: "yes",
+          primary_did_not_have_health_insurance: "yes",
+          spouse_did_not_have_health_insurance: "no",
+          authorize_sharing_of_health_insurance_info: nil
+        })
+        expect(form.valid?).to eq false
+      end
+
+      context "when sharing of health insurance info is authorized" do
+        context "when had_hh_member_without_health_insurance is yes" do
```

Comment:
i think these 2 blocks are unnecessary—the tests within can just be at this level ("had_hh_member_without_health_insurance is 'yes'". also it doesn't seem necessary for sharing of health insurance to be authorized, it just needs to have a value